src/shared/jsal.c

```diff
@@ -1034,6 +1034,12 @@ jsal_is_array(int stack_index)
 	return type == JsArray;
 }
 
+bool
+jsal_is_async_call(void)
+{
+	return s_async_flag;
+}
```

Comment:
How does this async detection work?

Reply:
There’s no actual detection; the flag is set based on whether the function was registered as async or not (see `jsal_push_new_function).  A call could even dynamically convert itself between async and sync it if wants since the promise-wrapping is done separately, after the native callback is finished.